hips/tiles/fetch.py

```diff
@@ -0,0 +1,120 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+import urllib.request
+import concurrent.futures
+from typing import Generator
+from ..tiles import HipsSurveyProperties, HipsTile, HipsTileMeta
+
+__all__ = [
+    'HipsTileFetcher',
+]
+
+
+class HipsTileFetcher:
+    """Fetch a list of HiPS tiles.
+
+    Parameters
+    ----------
+    tile_indices : np.ndarray
+        List of index values for HiPS tiles
+    hips_order : int
+        Order of the HiPS survey
+    hips_survey : `~hips.HipsSurveyProperties`
+        HiPS survey properties
+    tile_format : {'fits', 'jpg', 'png'}
+        Format of HiPS tile
+    progress_bar : bool
+        Show a progress bar for tile fetching and drawing
+    thread_count : int
```

Comment:
I see here in the `HipstileFetcher` you have an option `thread_count`. Please rename it to `n_parallel` or something that doesn't contain the word "thread", and then also use it to set the max number of parallel requests for the async fetching. I.e. this option should have an effect and work the same for both methods of fetching you implement.

Furthermore, the `timeout` should be listed here as well, and then again used in the same way for both methods to fetch.